core/src/software/aws/toolkits/core/ToolkitClientManager.kt

```diff
@@ -55,89 +52,98 @@ abstract class ToolkitClientManager {
         val serviceId = key.serviceClass.java.getField("SERVICE_METADATA_ID").get(null) as String
         if (serviceId !in GLOBAL_SERVICE_DENY_LIST && getRegionProvider().isServiceGlobal(connection.region, serviceId)) {
             val globalRegion = getRegionProvider().getGlobalRegionForService(connection.region, serviceId)
+            @Suppress("UNCHECKED_CAST")
             return cachedClients.computeIfAbsent(key.copy(region = globalRegion)) { createNewClient(sdkClass, connection.copy(region = globalRegion)) } as T
         }
 
+        @Suppress("UNCHECKED_CAST")
         return cachedClients.computeIfAbsent(key) { createNewClient(sdkClass, connection) } as T
     }
 
+    private fun <T : SdkClient> createNewClient(sdkClass: KClass<T>, connection: ConnectionSettings): T = constructAwsClient(
+        sdkClass = sdkClass,
+        credProvider = connection.credentials,
+        region = Region.of(connection.region.id),
+    )
+
+    /**
+     * Constructs a new low-level AWS client whose lifecycle is **NOT** managed centrally. Caller is responsible for shutting down the client
```

Comment:
are you envisioning this only being used for short-lived clients? what would be the best way to deal with long-lived clients

Reply:
Long lived too, like Telemetry uses it and thats the life of the toolkit. It just means that the callee is responsible for calling close on the client, not the toolkit on dispose